### ISSUE TITLE
IndexDefinition does not enforce 'mt_' prefix

### ISSUE THREAD

**Issue by jimgolfgti** [JasperFx/marten#534]
When overriding the name of an index if the index name does not start with 'mt_' then SchemaDiff will assume the index does not exist causing DocumentSchemaObjects to attempt to rebuild the "missing" index.

**Comment by jeremydmiller** [comment]
How do you think we should beat this? Could have Marten automatically prepend the "mt_" prefix, or throw a validation error if it's not there.

**Comment by jimgolfgti** [comment]
I've raised #536 which auto-appends the required prefix.

Throwing for this seemed a bit strong, this is probably a little used feature anyway.

I only became aware of it as we got bitten by the issue raised in #535 as the index name we used kept causing the "No document storage exists for type..." exception during schema validation. It wasn't immediately clear what the issue was which I suppose throwing during schema generation could have highlighted earlier.
If someone is overriding the index name they could be doing it so they can use the index from outside of Marten. Therefore if the name is manipulated transparently then users will not be aware of the change, however, there was already prior example of this in ComputedIndex so #536 followed.
